src/components/MemoDetail.vue

```diff
@@ -0,0 +1,67 @@
+<template>
+  <div class="backdrop" @click.self='closeDetail'>
+    <div class="modal">
+      <h2>Memo Detail</h2>
+      <label>
+        <input placeholder="Title" v-model='this.memo.title'>
+      </label>
+      <label>
+        <textarea placeholder="Content" v-model='this.memo.content'></textarea>
+      </label>
+      <button @click='editMemo()'>Edit</button>
+      <button @click='deleteMemo()'>Delete</button>
+    </div>
+  </div>
+</template>
+
+<script>
+export default {
+  name: 'MemoDetail',
+  data () {
+    const data = {
+      memos: [],
+      memo: {
+        id: '',
+        title: '',
+        content: ''
+      }
+    }
+    return data
+  },
+  mounted () {
+    if (localStorage.getItem('memos-vuejs')) {
+      this.memos = JSON.parse(localStorage.getItem('memos-vuejs'))
+      this.memo = this.memos.find((memo) => { return memo.id == this.$route.params.id })
```

Comment:
この場合はアロー関数の block と return を省略して書いたほうが読みやすいかもです！

```suggestion
      this.memo = this.memos.find((memo) => memo.id == this.$route.params.id)
```